fili-core/src/test/groovy/com/yahoo/bard/webservice/web/filters/RoleBasedAuthFilterSpec.groovy

```diff
@@ -15,49 +15,55 @@ import javax.ws.rs.core.SecurityContext
 @Timeout(30)
 class RoleBasedAuthFilterSpec extends Specification {
 
-    JerseyTestBinder jtb
+    private final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance()
 
-    Class<?>[] getResourceClasses() {
-        [RoleBasedAuthFilterSpec.class, DataServlet.class]
-    }
+    ContainerRequestContext containerRequestContext;
+    SecurityContext securityContext;
+    RoleBasedAuthFilter roleBasedAuthFilter;
 
     def setup() {
-        // Create the test web container to test the resources
-        jtb = new JerseyTestBinder(getResourceClasses())
-    }
+        containerRequestContext = Mock(ContainerRequestContext)
+        securityContext = Mock(SecurityContext)
+        containerRequestContext.getSecurityContext() >> securityContext
 
-    def cleanup() {
-        // Release the test web container
-        jtb.tearDown()
+        roleBasedAuthFilter = new RoleBasedAuthFilter()
+        SYSTEM_CONFIG.setProperty(SYSTEM_CONFIG.getPackageVariableName("user_roles"), "foo,bar");
     }
 
-    def "GET requests return false if the user is not in an allowed role"() {
-        given: "A request that does not have the security context of an user in role"
-        ContainerRequestContext context = Mock(ContainerRequestContext)
-        SecurityContext securityContext = Mock(SecurityContext)
-        securityContext.isUserInRole("foo") >> true
-        context.getSecurityContext() >> securityContext
-        List<String> allowedUserRoles = ["baz"]
+    def "GET request is not authorized if the user is not in an allowed role"() {
+        when: "received a request and that the user is not in the authorized roles"
+        containerRequestContext.getMethod() >> "GET"
+        securityContext.isUserInRole() >> false
```

Comment:
This mock won't work as you're expecting. It shouldn't ever trigger because it's not valid to call that method with no parameters. If you want the mock to trigger on all calls to that method, you should use the wildcard parameter syntax: `securityContext.isUserInRole(*_)`